### DISCUSSION TITLE
Througput is not increasing by increasing worker count

### DISCUSSION THREAD

**Discussion by msenger1987** [Farfetch/kafkaflow#273] [Q&A]
My application is a kafka topic consumer. After reading msg from topic it calls 5 external api for each message. 
For each msg external APIs are taking 225 ms.
Topic having 2 partitions.
Application current throughput is 700 msg/min but requirement is to increase throughput of app to 5000 msg/min.
Use used KafkaFlow but even after Worker count 10 desired throughput is not acheived.
Can any one help here to identify what could be done to acheive desired throughput.

**Comment by msenger1987** [comment]
@farfetch-automator Can you please suggest any pointer for above problem

**Comment by msenger1987** [comment]
[filipeesch](https://github.com/filipeesch) and [dougolima](https://github.com/dougolima) Can you please suggest any pointer for above problem

**Comment by filipeesch** [comment]
Hi @msenger1987,

When you increase the parallelism you put more load on your application dependencies (in your case, on the external APIs) and, sometimes, they start to respond slower. You have to find the sweet spot of parallelism for your application and try to optimize these external calls (maybe some of these calls can be made in parallel).
